packages/psss-server/src/connections/MeditrakConnection.ts

```diff
@@ -3,10 +3,112 @@
  * Copyright (c) 2017 - 2020 Beyond Essential Systems Pty Ltd
  */
 
+import { generateId } from '@tupaia/database';
+import { convertPeriodStringToDateRange } from '@tupaia/utils';
 import { ApiConnection } from './ApiConnection';
 
 const { MEDITRAK_API_URL = 'http://localhost:8090/v2' } = process.env;
 
+const SURVEY_RESPONSE_FIELDS = <const>['entity.code', 'survey.code', 'submission_time', 'id'];
+const ANSWER_FIELDS = <const>['question.code', 'id'];
+const PSSS_SURVEY_RESPONSE_ANSWER_TYPE = 'Number'; // All survey response answers are 'Number' in PSSS
+
+type SurveyResponseObject = {
+  [key in typeof SURVEY_RESPONSE_FIELDS[number]]: string;
+};
+
+type AnswerObject = {
+  [key in typeof ANSWER_FIELDS[number]]: string;
+};
+
 export class MeditrakConnection extends ApiConnection {
   baseUrl = MEDITRAK_API_URL;
+
+  async findSurveyResponse(surveyCode: string, orgUnitCode: string, period: string) {
+    const [startDate, endDate] = convertPeriodStringToDateRange(period);
+    const results = (await this.get(`surveyResponses/`, {
+      page: '0',
+      pageSize: '1',
+      columns: `["${SURVEY_RESPONSE_FIELDS.join('","')}"]`,
+      filter: JSON.stringify({
+        'survey.code': { comparisonValue: surveyCode, castAs: 'text' },
+        'entity.code': { comparisonValue: orgUnitCode, castAs: 'text' },
```

Comment:
Are these casts necessary?

Reply:
I am not sure, I copied them from the admin panel, I'll check 👍

Reply:
They were not necessary, so I removed them